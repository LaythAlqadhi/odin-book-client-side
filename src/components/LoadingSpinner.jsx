import React from 'react';

function LoadingSpinner() {
  return (
    <div data-testid="loading" className="flex h-screen w-full items-center justify-center">
      <div className="h-20 w-20 animate-spin rounded-full border-8 border-gray-200 border-t-blue-500" />
    </div>
  );
}

export default LoadingSpinner;
