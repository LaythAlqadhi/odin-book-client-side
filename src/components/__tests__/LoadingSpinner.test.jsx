import React from 'react';
import { render, screen } from '@testing-library/react';
import LoadingSpinner from '../LoadingSpinner';

describe('LoadingSpinner component', () => {
  it('renders LoadingSpinner component correctly', () => {
    render(<LoadingSpinner />);

    expect(screen.container).toMatchSnapshot();
  });
});
