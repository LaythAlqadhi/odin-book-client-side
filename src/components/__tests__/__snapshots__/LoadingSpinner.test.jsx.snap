// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner component > renders LoadingSpinner component correctly 1`] = `undefined`;
